libbpfgo.go

```diff
@@ -531,6 +531,35 @@ func CreateMap(mapType MapType, mapName string, keySize, valueSize, maxEntries i
 	}, nil
 }
 
+// GetMapByID returns a BPF map by ID. This can be used for reaching the maps
+// from a BPF array of maps or hash of maps. However, this function uses a low-level
+// libbpf API; maps loaded in this way do not conform to libbpf map formats,
+// and therefore do not have access to libbpf high level bpf_map__* APIS
+// which causes different behavior from maps created in the kernel side code.
+//
+// See usage of `bpf_map_get_fd_by_id()` in kernel selftests for more info.
+func (m *Module) GetMapByID(id uint32) (*BPFMap, error) {
```

Comment:
I'll promote this to package level, since the BPFMap returned doesn't have libbpf module and map pointers.